dashboard/app/views/levels/_project_ui.html.haml

```diff
@@ -0,0 +1,103 @@
+- if @level.properties['free_play'] == 'true'
+  %a#create-project{href: '#'} Create project
+  %span#project-name
+  %span#save-project
+  %form#project-form{style: 'display: none'}
+    Rename:
+    %input#name{type: 'text'}
+    %input{type: 'submit'}
+- else
+  Not freeplay, cannot create project
+
+%script{src: '/shared/js/apps_api.js'}
+
+:javascript
+  var apps = storageApps();
+  var allApps = {};
+  var currentApp = {};
+
+  var displayCurrentApp = function() {
+    console.log(currentApp);
+    $('#create-project').hide();
+    $('#project-name').text(currentApp.name);
+    $('#project-form input#name').val(currentApp.name);
+    $('#project-form').show();
+    $('#save-project').show();
+  }
+
+  $(document).ready( function() {
+    if (window.location.hash && window.location.hash.length > 1) {
+      // try to load a project from the hash param
+      apps.fetch(window.location.hash.slice(1),
+        function(data) {
+          if (data) {
+            currentApp = data;
+            console.log("fetched");
+            displayCurrentApp();
+          } else {
+            console.log('failed to fetch project ' + window.location.hash);
+          }
+        });
+    }
+  });
+
+  $(document).bind('ajaxSuccess', function(event, response) {
```

Comment:
Binding to `ajaxSuccess` will fire this callback on *all* successful AJAX responses.  For example this throws an exception when opening the user stats overlay from the header.